### PR TITLE
bumpver: init at 2020.1107 + python.pkgs.lexid

### PR DESCRIPTION
**Pull request by kfollesdal**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pycalver</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10444,6 +10444,8 @@ in
   };
   purePackages = recurseIntoAttrs (callPackage ./pure-packages.nix {});
 
+  pycalver = callPackage ../applications/version-management/pycalver { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10447]: This position is wrong. Around line 6667 should be right.

File: pkgs/applications/version-management/pycalver/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "6a194b9b14c8c81a7dbefb0659de4a156c49bf513f3bdb1aba49b371fff778d9";
+  };
+
+  checkInputs = [ python3.pkgs.pytestCheckHook git mercurial];
+  pytestFlagsArray = [ "--ignore=test/test_config.py" "--ignore=test/test_rewrite.py"];
+
+  propagatedBuildInputs = with python3.pkgs; [ pathlib2 click toml six setuptools ];
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/pycalver/default.nix:15]: Can you put this before checkInputs? Build process happens logically there.

File: pkgs/applications/version-management/pycalver/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with lib; {
+    description = "Bump version numbers in project files";
+    homepage = "https://pypi.org/project/pycalver/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ kfollesdal ];
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/pycalver/default.nix:23]:
```suggestion
}

```

Missing final new line.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pycalver</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pycalver</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (kfollesdal) [reply pkgs/top-level/all-packages.nix:10447]: Thanks, was not sure about position.

**Comment by kfollesdal** [comment]
Thanks @SuperSandro2000 for good comments. Have now updated with your improvements.

**Comment by kfollesdal** [comment]
One moment. Just come out new version with a name change. Make a update now.

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python27Packages.lexid</li>
    <li>python27Packages.lib3to6</li>
    <li>python37Packages.lib3to6</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
    <li>python38Packages.lib3to6</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python27Packages.lexid</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
@SuperSandro2000 I have then got the new version to work. Had to add a missing python dependency `lexid`.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -3361,6 +3361,7 @@ in {
 
   leveldb = callPackage ../development/python-modules/leveldb { };
 
+  lexid = callPackage ../development/python-modules/lexid { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3364]:
```suggestion
  lexid = callPackage ../development/python-modules/lexid { };
  
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (kfollesdal) [reply pkgs/top-level/python-packages.nix:3364]: Thanks @SuperSandro2000

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

File: pkgs/development/python-modules/lexid/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib, python, pythonOlder, buildPythonPackage, fetchPypi, pytestCheckHook, click }:
+
+buildPythonPackage rec {
+  pname = "lexid";
+  version = "2020.1005";
+  disabled = pythonOlder "3.6";
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lexid/default.nix:8]:
```suggestion

```

File: pkgs/development/python-modules/lexid/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    inherit pname version;
+    sha256 = "52333a2b9ebd14aa0dfeb33de72bd159c2dc31adb9c59cddfc486e2b69bfdcd1";
+  };
+
+  propagatedBuildInputs = [ click ];
+
+  checkInputs = [ pytestCheckHook ];
+
+  meta = with lib; {
+    description = "lexid is a micro library to increment lexically ordered numerical ids";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lexid/default.nix:19]:
```suggestion
    description = "micro library to increment lexically ordered numerical ids";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
  </ul>
</details>

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Can you please squash the commits into one commit for each package? Thanks.

**Comment by kfollesdal** [comment]
Thanks @SuperSandro2000 for good feedback. Have now squash the commits on to one for each package. I did not do this right away so you easy could see the changes.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
    <li>python39Packages.lexid</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bumpver</li>
    <li>python37Packages.lexid</li>
    <li>python38Packages.lexid</li>
    <li>python39Packages.lexid</li>
  </ul>
</details>

**Pull request merged**